src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/BOM/MaterialForm/MaterialFormView.tsx

```diff
@@ -202,23 +205,27 @@ const MaterialFormView: React.FC<MaterialFormViewProps> = ({
                     variant="outlined"
                     error={!!errors.unitName}
                     helperText={errors.unitName?.message}
+                    value={field.value || ''}
+                    onChange={(event) => {
+                      const selectedValue = event.target.value;
+                      if (selectedValue === 'createUnit') {
+                        const unitName = prompt('Enter Unit Name');
+                        if (unitName) {
+                          createUnit(unitName);
+                        } else {
```

Comment:
what is this else doing in both cases?

Reply:
If the user cancels the prompt and unitName is undefined. The it will change the event's value to the previous value (field.value). I just pushed another commit that removed the else case for when the selected value is not "createUnit" since it was unnecessary, since the field should be changed when the text field changes.